### PR TITLE
Add ut for execution controller

### PR DESCRIPTION
**Pull request by jwcesign**
Signed-off-by: jwcesign <jiangwei115@huawei.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Add ut for execution controller

**Which issue(s) this PR fixes**:
Part of https://github.com/karmada-io/karmada/issues/2579

**Special notes for your reviewer**:
none

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

**Comment by jwcesign** [comment]
cc @Poor12

File: pkg/controllers/execution/execution_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+	if err != nil {
+		t.Fatalf("error:%v", err)
+	}
+	err = clusterv1alpha1.AddToScheme(scheme)
+	if err != nil {
+		t.Fatalf("error:%v", err)
+	}
+
+	restMapper := meta.NewDefaultRESTMapper(scheme.PreferredVersionAllGroups())
+	workGVR := schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"}
```
Comment (Poor12) [inline pkg/controllers/execution/execution_controller_test.go:48]: deploymentGVR

**Review by Poor12 — COMMENTED** [review]
Generally LGTM

File: pkg/controllers/execution/execution_controller_test.go (hunk 1 above)
Comment (jwcesign) [reply pkg/controllers/execution/execution_controller_test.go]: fixed

**Comment by jwcesign** [comment]
cc @RainbowMango

File: pkg/util/fedinformer/genericmanager/multi_cluster_manager.go [hunk 2]
```diff
@@ -1,3 +1,5 @@
+//go:generate mockgen -source=multi_cluster_manager.go  -destination=testing/mock_multi_cluster_manager.go -package=testing MultiClusterInformerManager
```
Comment (chaunceyjiang) [inline pkg/util/fedinformer/genericmanager/multi_cluster_manager.go:1]: Just a question, Do I need to install the `mockgen` command in advance?

Comment (jwcesign) [reply pkg/util/fedinformer/genericmanager/multi_cluster_manager.go:1]: Yes, this PR too big, check this one first: https://github.com/karmada-io/karmada/pull/2654

**Comment by RainbowMango** [comment]
ping @jwcesign 
If this is still needed, please solve the conflicts.

**Comment by jwcesign** [comment]
The controller unit test is too difficult to maintain, so let's drop it.

**Pull request closed without merging**